be/src/vec/functions/like.cpp

```diff
@@ -253,7 +255,8 @@ Status FunctionLikeBase::constant_starts_with_fn_scalar(LikeSearchState* state,
                                                         const StringRef& pattern,
                                                         unsigned char* result) {
     *result = (val.size >= state->search_string_sv.size) &&
-              (state->search_string_sv == val.substring(0, state->search_string_sv.size));
+              (state->search_string_sv ==
+               val.substring(0, cast_set<int>(state->search_string_sv.size)));
```

Comment:
why consider change the `substring` api to int64

Reply:
fixed